protocol/storage/storagedeal/state.go

```diff
@@ -1,13 +1,15 @@
-package storage
+package storagedeal
```

Comment:
blocking: why do we have a new package for this? i suspect an import cycle which speaks to a deeper problem. i think the issue is likely that storage protocol depends on deal put/ls plumbing, which depend on storage protocol. this is a strong signal that deal put/ls do not belong in plumbing. the putter and getter should be direct, private dependencies of the storage protocol and should not be in the plumbing interface. commands that need deal ls should get it from the deal lser the storage protocol uses, which can be surfaced to them in storage protocol control api which will be separate from plumbing because it is higher level, which is part of https://github.com/filecoin-project/go-filecoin/issues/2030. for now we can keep it but @shannonwells can you please ack (and @rkowalick please ensure she does) that undoing this -- moving the deal back into the storage package as storage.deal and removing deal put/ls from plumbing -- is part of the scope for 2030? sorry to ramble here, i'm going ooo. and sorry @rkowalick this is not really something you could have forseen, we are underprepared to move protocol specific calls at this point, we'll have to play catch up.

Reply:
Yes it makes sense to have everything that's interacting with storage deals use the upcoming protocol api, if that's what you're asking.

Reply:
@phritz I agree entirely that this is the core of the problem. It seems the protocol refactor stuff that @shannonwells will likely reorganize this stuff, so it seems like wasted effort to pull this stuff out of plumbing right now.